mautic/mautic
# Fix scaffolded files and add sanity checks to github workflow

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR addresses steps in the QA process that are easily forgotten:
- the `phpstan-baseline.neon` should only shrink, not altered (see [the commits after Nov 9, 2021](https://github.com/mautic/mautic/commits/4.x/phpstan-baseline.neon) for recent changes)
- the files used for scaffolding should be in sync with the original ones (e.g. the [original `.gitignore`](https://github.com/mautic/mautic/commits/4.x/.gitignore) has commits not present in the  [scaffolded `.gitignore`](https://github.com/mautic/mautic/commits/4.x/app/assets/scaffold/files/example.gitignore))



#### Steps to test this PR:

The tests in this PR fail intentionally to show the effect:
see https://github.com/mautic/mautic/pull/10857/checks
* the [`scaffolded files mismatch`](https://github.com/mautic/mautic/runs/5172853079?check_suite_focus=true#logs) fails when the scaffolded files that are not in sync with the original ones
* the [`PHPStan baseline changes`](https://github.com/mautic/mautic/runs/5172853097?check_suite_focus=true#logs) fails when there are unwanted changes to the `phpstan-baseline.neon` file

Once this makes sense to the reviewers of this PR, following steps are needed **before merging**:
* fix the scaffolded files
* revert the commit(s) that alter the `phpstan-baseline.neon` files


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10857"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
